src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -535,16 +571,118 @@ def _visit_neighbors(self, node: gtir.FunCall) -> DataExpr:
             memlet=dace.Memlet(data=neighbors_temp, subset=neighbor_idx),
         )
 
-        assert isinstance(node.type, itir_ts.ListType)
         return DataExpr(neighbors_node, node.type)
 
+    def _visit_map(self, node: gtir.FunCall) -> DataExpr:
+        """
+        A map node defines an operation to be mapped on all elements of input arguments.
+
+        The map operation is applied on the local dimension of input fields.
+        In the example below, the local dimension consists of a list of neighbor
+        values as the first argument, and a list of constant values `1.0`:
+        `map_(plus)(neighbors(V2Eₒ, it), make_const_list(1.0))`
+
+        The `plus` operation is lowered to a tasklet inside a map that computes
+        the domain of the local dimension (in this example, the number of neighbors).
+        """
+        assert isinstance(node.type, itir_ts.ListType)
+        assert isinstance(node.fun, gtir.FunCall)
+        assert len(node.fun.args) == 1  # the operation to be mapped on the arguments
+
+        assert isinstance(node.type.element_type, ts.ScalarType)
+        dtype = dace_utils.as_dace_type(node.type.element_type)
+
+        input_args = [self.visit(arg) for arg in node.args]
+        connectors = [f"__arg{i}" for i in range(len(input_args))]
+
+        # Convert the operation to be mapped into a function call on a list of
+        # internal parameters with the same names as the tasklet connectors.
+        fun_node = im.call(node.fun.args[0])(*connectors)
+        op_code = gtir_python_codegen.get_source(fun_node)
+
+        # TODO(edopao): extract offset_dim from the input arguments
+        offset_dim = gtx_common.Dimension("", gtx_common.DimensionKind.LOCAL)
+        map_index = dace_gtir_utils.get_map_variable(offset_dim)
+
+        # The dataflow we build in this class has some loose connections on input edges.
+        # These edges are described as set of nodes, that will have to be connected to
+        # external data source nodes passing through the map entry node of the field map.
+        # For `map_` and `neighbors` expressions, these edges will terminate on the view
+        # nodes (see the for-loop below), because it is simpler than representing
+        # map-to-map edges (which require memlets with 2 pass-nodes).
+        input_memlets = {}
+        input_nodes = {}
+        local_size: Optional[int] = None
+        for conn, input_expr in zip(connectors, input_args, strict=True):
+            if isinstance(input_expr, MemletExpr):
+                if set(input_expr.subset.size()[:-1]) != {1}:
+                    raise ValueError(f"Invalid node {node}")
+                rstart, rstop, rstep = input_expr.subset[-1]
+                assert rstart == 0 and rstep == 1
+
+                desc = self.sdfg.arrays[input_expr.node.data]
```

Comment:
I would add a note to inform the reader that in the current IR the local dimension is always at the end.

Please also add the TODO:
```
TODO(phimuell): Ensure that the local memory layout is correct/corrected during optimization
```